core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -0,0 +1,183 @@
+      // TODO(bhenning): This can be made far more optimal once individual
+      // prerequisite and acquired skill lists are also stored in the skill list
+      // frontend domain object.
+      $scope.collectionSkillList.clearSkills();
+      for (var i = 0; i < $scope.collection.nodes.length; i++) {
+        var collectionNode = $scope.collection.nodes[i];
+        var prerequisiteSkills = SkillListObjectFactory.create(
+          collectionNode.prerequisite_skills);
+        var acquiredSkills = SkillListObjectFactory.create(
+          collectionNode.acquired_skills);
+        $scope.collectionSkillList.concatSkillList(prerequisiteSkills);
+        $scope.collectionSkillList.concatSkillList(acquiredSkills);
+      }
+      $scope.collectionSkillList.sortSkills();
+    };
+
+    // Stores a pending list of changes.
+    $scope.changeList = [];
+
+    $scope.addChange = function(change) {
+      $scope.changeList.push(change);
+    };
+
+    $scope.addPrerequisiteSkill = function(skillName, explorationId) {
+      var collectionNode = _getCollectionNodeByExplorationId(explorationId);
+      var prerequisiteSkills = collectionNode.prerequisite_skills;
+      prerequisiteSkills.push(skillName);
+      _resetSkillList();
+    };
+
+    $scope.deletePrerequisiteSkill = function(skillIndex, explorationId) {
+      var collectionNode = _getCollectionNodeByExplorationId(explorationId);
+      var prerequisiteSkills = collectionNode.prerequisite_skills;
+      var skillName = prerequisiteSkills[skillIndex];
+      prerequisiteSkills.splice(skillIndex, 1);
+      _resetSkillList();
+    };
+
+    $scope.addAcquiredSkill = function(skillName, explorationId) {
+      var collectionNode = _getCollectionNodeByExplorationId(explorationId);
+      var acquiredSkills = collectionNode.acquired_skills;
+      acquiredSkills.push(skillName);
+      _resetSkillList();
+    };
+
+    $scope.deleteAcquiredSkill = function(skillIndex, explorationId) {
+      var collectionNode = _getCollectionNodeByExplorationId(explorationId);
+      var acquiredSkills = collectionNode.acquired_skills;
+      var skillName = acquiredSkills[skillIndex];
+      acquiredSkills.splice(skillIndex, 1);
+      _resetSkillList();
+    };
+
+    $scope.deleteCollectionNode = function(explorationId) {
+      for (var i = 0; i < $scope.collection.nodes.length; i++) {
+        var collectionNode = $scope.collection.nodes[i];
+        if (collectionNode.exploration_id == explorationId) {
+          $scope.collection.nodes.splice(i, 1);
+          return;
+        }
+      }
+      warningsData.fatalWarning(
+        'Internal collection editor error. Could not delete exploration by ' +
+        'ID: ' + explorationId);
+    };
+
+    $scope.hasCollectionNode = function(explorationId) {
+      return _getCollectionNodeByExplorationId(explorationId) != null;
+    };
+
+    $scope.addCollectionNode = function(explorationId) {
+      var newNode = {
+        exploration_id: explorationId,
```

Comment:
This was intentional to match a pattern started several PRs ago when the backed constructs the learner dict used by both the collection player and editor. This pattern seems unused (at least after the latest refactor), so I've removed it entirely.